Kernel/FileSystem/Ext2FileSystem.cpp

```diff
@@ -1508,7 +1508,8 @@ KResultOr<NonnullRefPtr<Inode>> Ext2FS::create_inode(InodeIdentifier parent_id,
     static_cast<Ext2FSInode&>(*inode).m_block_list = move(blocks);
 
     auto result = parent_inode->add_child(*inode, name, mode);
-    ASSERT(result.is_success());
+    if (result.is_error())
+        return result;
```

Comment:
At this point we've already allocated and written an inode to disk, let's do the check and fail earlier. There's no need to do any I/O if the filename is simply too long :)